routes/index.js

```diff
@@ -326,7 +326,7 @@ router.get('/dashboard/:study/:subject', ensurePermission, function(req, res) {
             } else if (!doc || Object.keys(doc).length === 0) {
                 return res.status(404).send({});
             } else {
-                if('config' in doc['preferences']) {
+                if(doc.preferences && 'config' in doc.preferences) {
```

Comment:
@tashrifbillah  🐛 Fix /dashboard/:study/:subject undef check

Reply:
Good check!

Reply:
Basically doc['preferences'] was undefined, so the `for...in` statement was throwing an error. I also prefer using dot notation for referring to keys in objects, unless the square bracket notation is necessary (such as characters like `-` appearing in the key name).

Reply:
Sure!

Reply:
> (such as characters like - appearing in the key name)

Or space. I believe that is why everywhere we have used `[ ... ]`. I do not have a problem with you using the `.` notation but think if it would harm consistency of style in the codebase.

Reply:
That makes sense. Easy enough to change back.

Reply:
Fixed in 6781c494f14c202d5ece263df8ccf2cc5b31c68f.